tests/test_input_outputs.py

```diff
@@ -251,3 +254,70 @@ def test_BaseWrapper_postprocess_y():
     np.testing.assert_equal(y_postprocessed, y_array)
     extra_args = BaseWrapper.postprocess_y(y_array)[1]
     assert len(extra_args) == 0
+
+
+@pytest.mark.parametrize(
+    "X_dtype", ["float32", "float64", "int64", "int32", "uint8", "uint16"]
+)
+@pytest.mark.parametrize("y_dtype", ["int64", "int32", "uint8", "uint16"])
+@pytest.mark.parametrize("s_w_dtype", ["float32", "float64"])
+@pytest.mark.parametrize("run_eagerly", [True, False])
+def test_classifier_handles_types(X_dtype, y_dtype, s_w_dtype, run_eagerly):
+    """Tests that classifiers correctly handle dtype conversions and
+    return the same dtype as the inputs.
+    """
+    n, d = 100, 20
+    n_classes = 10
+    X = np.random.uniform(size=(n, d)).astype(X_dtype)
+    y = np.random.choice(n_classes, size=n).astype(y_dtype)
+    sample_weight = np.random.random(y.shape).astype(s_w_dtype)
+
+    input_dtypes = (X.dtype, y.dtype, sample_weight.dtype)
+
+    class StrictClassifier(KerasClassifier):
+        def _fit_keras_model(self, X, y, sample_weight, warm_start, **kwargs):
+            assert X.dtype == input_dtypes[0]
+            assert y.dtype == input_dtypes[1]
+            assert sample_weight.dtype == input_dtypes[2]
+            return super()._fit_keras_model(
+                X, y, sample_weight, warm_start, **kwargs
+            )
+
+    clf = StrictClassifier(
+        build_fn=dynamic_classifier, run_eagerly=run_eagerly
+    )
+    clf.fit(X, y, sample_weight=sample_weight)
+    assert clf.score(X, y) >= 0
+    assert clf.predict(X).dtype == y.dtype
+
+
+@pytest.mark.parametrize(
+    "X_dtype", ["float32", "float64", "int64", "int32", "uint8", "uint16"]
+)
+@pytest.mark.parametrize("y_dtype", ["float32", "float64"])
+@pytest.mark.parametrize("s_w_dtype", ["float32", "float64"])
```

Comment:
Can we keep tests for other int dtypes?

Reply:
How is an integer output array `y` relevant for a regression problem? I think floating point values are far more relevant.

Reply:
Output Is not relevant. Outputs should always be float. But it should accept int inputs.